monai/transforms/spatial/array.py

```diff
@@ -340,14 +229,6 @@ class ResampleToMatch(SpatialResample):
     """Resample an image to match given metadata. The affine matrix will be aligned,
     and the size of the output image will match."""
 
-    def update_meta(self, img: torch.Tensor, dst_affine=None, img_dst=None):
-        if dst_affine is not None:
-            super().update_meta(img, dst_affine)
```

Comment:
Hi @Nic-Ma, do you mean save meta information in a dictionary like `meta_dict`? I think the same logic is retained here as in the original.

Reply:
Hi @KumoLiu ,

I think the original logic will update meta no matter the image is `MetaTensor` or not, but the changed logic of this PR will only update meta when `if isinstance(img, MetaTensor)` as line 280, right?

Thanks.

Reply:
Hi @Nic-Ma, the original logic also only update affine when data is `MetaTensor` since we didn't have a meta dict before. If data is `non-MetaTensor`, affine information would be lost. Or did I misunderstand it?
https://github.com/Project-MONAI/MONAI/blob/b9e17e8186cb25c8781e2c82de3b4915081df48b/monai/transforms/spatial/array.py#L189-L190

Reply:
OK, seems we have bug in the dev, if input is not MetaTensor, it still updates meta.
Thanks for the fix!